src/test/java/org/matsim/urbanEV/UrbanEVTests.java

```diff
@@ -95,21 +104,635 @@ public void testAgentsExecuteSameNumberOfActs(){
+	}
+
+	@Test
+	public void testNotEnoughTimeCharger(){
+		List<ActivityStartEvent> plugins = this.handler.plugInCntPerPerson.get(Id.createPersonId("Not enough time Charger"));
+		Assert.assertEquals(1, plugins.size(),0);
+		ActivityStartEvent pluginActStart = plugins.get(0);
+		Assert.assertEquals("wrong charging start time",  24940d, pluginActStart.getTime(), MatsimTestUtils.EPSILON);
+		Assert.assertEquals("wrong charging start location", "89", pluginActStart.getLinkId().toString());
+
+		List<ActivityEndEvent> plugouts = this.handler.plugOutCntPerPerson.get(Id.createPersonId("Not enough time Charger"));
+		Assert.assertEquals( 1, plugouts.size(),0);
+		ActivityEndEvent plugoutActStart = plugouts.get(0);
+		Assert.assertEquals("wrong charging end time",  47729d, plugoutActStart.getTime(), MatsimTestUtils.EPSILON);
+		Assert.assertEquals("wrong charging end location", "89", plugoutActStart.getLinkId().toString());
 	}
 
+	@Test
+	public void testHomeCharger(){
+		List<ActivityStartEvent> plugins = this.handler.plugInCntPerPerson.get(Id.createPersonId("Home Charger"));
+		Assert.assertEquals(1, plugins.size(),0);
+		ActivityStartEvent pluginActStart = plugins.get(0);
+		Assert.assertEquals("wrong charging start time",  6225d, pluginActStart.getTime(), MatsimTestUtils.EPSILON);
+		Assert.assertEquals("wrong charging start location", pluginActStart.getLinkId().toString(), "91");
+
+		List<ActivityEndEvent> plugouts = this.handler.plugOutCntPerPerson.get(Id.createPersonId("Home Charger"));
+		Assert.assertEquals("Home charge should not have a plug out interaction",  0, plugouts.size());
+	}
+
+	@Test
+	public void testDoubleChargerHomeCharger(){
+		List<ActivityStartEvent> plugins = this.handler.plugInCntPerPerson.get(Id.createPersonId("Double Charger Home Charger"));
+		Assert.assertEquals(plugins.size(),2,0);
+		ActivityStartEvent pluginActStart = plugins.get(0);
+		Assert.assertEquals("wrong charging start time", pluginActStart.getTime(), 19683d, MatsimTestUtils.EPSILON);
+		Assert.assertEquals("wrong charging start location", "91", pluginActStart.getLinkId().toString());
+
+		ActivityStartEvent pluginActStart2 = plugins.get(1);
+		Assert.assertEquals("wrong charging start time", pluginActStart2.getTime(), 38444d, MatsimTestUtils.EPSILON);
+		Assert.assertEquals("wrong charging start location", "90", pluginActStart2.getLinkId().toString());
+
+
+		List<ActivityEndEvent> plugouts = this.handler.plugOutCntPerPerson.get(Id.createPersonId("Double Charger Home Charger"));
+		ActivityEndEvent plugoutActStart = plugouts.get(0);
+		Assert.assertEquals("wrong charging end time", 23891d, plugoutActStart.getTime(), MatsimTestUtils.EPSILON);
+		Assert.assertEquals("wrong charging end location", "91", plugoutActStart.getLinkId().toString());
+		Assert.assertEquals("Should only plug out once ",1, plugouts.size());
+	}
+
+	private static void overridePopulation(Scenario scenario) {
```

Comment:
@Jonas116 can you maybe create a population.xml (just write them out once before they are run in this test here) and instead of override plans per code rather provide the input plans in the config such that they are read in. This way, we would save hundreds of lines of code. Please put the inputPlans into the test input folder.